plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/expression/RewriteComparison.java

```diff
@@ -0,0 +1,143 @@
+        NOT_EQUAL(NOT_EQUAL_OPERATOR_FUNCTION_NAME, "$ne"),
+        LESS_THAN(LESS_THAN_OPERATOR_FUNCTION_NAME, "$lt"),
+        LESS_THAN_OR_EQUAL(LESS_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME, "$lte"),
+        GREATER_THAN(GREATER_THAN_OPERATOR_FUNCTION_NAME, "$gt"),
+        GREATER_THAN_OR_EQUAL(GREATER_THAN_OR_EQUAL_OPERATOR_FUNCTION_NAME, "$gte"),
+        /**/;
+
+        private final FunctionName functionName;
+        private final String operator;
+
+        private static final Map<FunctionName, ComparisonOperator> OPERATOR_BY_FUNCTION_NAME = Stream.of(values())
+                .collect(toImmutableMap(ComparisonOperator::getFunctionName, identity()));
+
+        ComparisonOperator(FunctionName functionName, String operator)
+        {
+            this.functionName = requireNonNull(functionName, "functionName is null");
+            this.operator = requireNonNull(operator, "operator is null");
+        }
+
+        private FunctionName getFunctionName()
+        {
+            return functionName;
+        }
+
+        private String getOperator()
+        {
+            return operator;
+        }
+
+        private static ComparisonOperator forFunctionName(FunctionName functionName)
+        {
+            return verifyNotNull(OPERATOR_BY_FUNCTION_NAME.get(functionName), "Function name not recognized: %s", functionName);
+        }
+    }
+
+    private final Pattern<Call> pattern;
+
+    public RewriteComparison()
+    {
+        Set<FunctionName> functionNames = Arrays.stream(ComparisonOperator.values())
+                .map(ComparisonOperator::getFunctionName)
+                .collect(toImmutableSet());
+
+        pattern = call()
+                .with(functionName().matching(functionNames::contains))
+                .with(type().equalTo(BOOLEAN))
+                .with(argumentCount().equalTo(2))
+                .with(argument(0).matching(expression().capturedAs(LEFT)))
+                .with(argument(1).matching(expression().capturedAs(RIGHT)));
+    }
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return pattern;
+    }
+
+    @Override
+    public Optional<MongoExpression> rewrite(Call call, Captures captures, RewriteContext<MongoExpression> context)
+    {
+        Optional<MongoExpression> left = context.defaultRewrite(captures.get(LEFT));
+        if (left.isEmpty()) {
+            return Optional.empty();
+        }
+        Optional<MongoExpression> right = context.defaultRewrite(captures.get(RIGHT));
+        if (right.isEmpty()) {
+            return Optional.empty();
+        }
+        verify(call.getFunctionName().getCatalogSchema().isEmpty()); // filtered out by the pattern
+        Object leftValue = left.get().expression();
+        Object rightValue = right.get().expression();
+
+        ComparisonOperator operator = ComparisonOperator.forFunctionName(call.getFunctionName());
+        if (captures.get(LEFT) instanceof Variable) {
+            leftValue = "$" + leftValue;
```

Comment:
Is it possible to move it to a static class ?

Adding a `$` is only for specific operations or ?

Reply:
This part may be not required if I place `$` while rewriting the variable in `RewriteVariable`.